packages/runtime/lib/runtime.js

```diff
@@ -506,6 +510,23 @@ export class Runtime extends EventEmitter {
     this.emit('application:stopped', id)
   }
 
+  async restartApplication (id) {
+    const config = this.#config
+    const applicationConfig = this.#config.applications.find(s => s.id === id)
+    const workersCount = await this.#workers.getCount(id)
+
+    this.emit('application:restarting', id)
+
+    for (let i = 0; i < workersCount; i++) {
+      const label = `${id}:${i}`
+      const worker = this.#workers.get(label)
+
+      await this.#replaceWorker(config, applicationConfig, workersCount, id, i, worker, true)
```

Comment:
Can we consider adding a configurable timeout here - after replacing the worker?

Use case: this can help to ease and distribute potential latency impact from new workers while restarting a running application - if there's no separate warm up step, new connections will take extra time establish, local caches, JIT, etc.

Reply:
In general, the internal logic for worker starting is subject to the same timeout of when Watt is started. This means that the new worker will be terminated if not started in time and the old one will never be replace.
The entire process will both throw an error and emit an event.

But I'm afraid I'm not understanding what you're asking for, can you please expand this?

Reply:
Sure, I meant an optional timeout before we begin restarting the next worker, not the graceful exit timeout.

In pseudo-code:
```js
      const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
      
      const label = `${id}:${i}`
      const worker = this.#workers.get(label)
      
      await this.#replaceWorker(config, applicationConfig, workersCount, id, i, worker, true)
      if (config.restartInterval) {
        await sleep(config.restartInternval)
      }
```

Some apps may be inefficient at start and may have higher latency at start. In this case, if we restart all workers very quickly, there might be a visible latency impact - trigger alerts / SLO violations on worker restart.